### ISSUE TITLE
Error "The disk is in use or locked by another process"

### ISSUE THREAD

**Issue by StefanPlizga** [microsoftarchive/redis#309]
Hello,
I sometimes get this error on slave instance, using latest build Redis-X64-2.8.2102, but the issue was also there with older releases.
I get this error by using redis-cli, and get another type of error when trying to connect with C# client from StackExchange, but it looks like the INFO command can't be issued at all in this case.
The only way to recover from this is to restart the slave instance. What could cause this error?
Thanks, Stefan

**Comment by enricogior** [comment]
Hi @StefanPlizga 
does the error message appear in the server log or is shown by the client?
In Redis on Windows there are Posix error codes and Windows error codes, sometimes the mapping from the error code to the error message is wrong, and this may be one of those cases.
Does this error occur when you issue a particular command from the client or is it random?
If the error is in the log can you post a few lines before and after the error to see if there is any clue to what is going on?
Thank you.

**Comment by StefanPlizga** [comment]
Hi @enricogior,
Actually, I get this error message when using redis-cli and issuing the INFO command on the slave.
The .NET StackExchange redis client does display another error message when retrieving information. Actually it looks like replication between the master and the slave works fine, however I don't know if reading from the slave is possible in this situation (I would need to test when it occurs again). 
This is just a bit bizarre that redis-cli is not able to retrieve data from the slave in this situation, as I am wondering is redis is working as expected or not.

**Comment by enricogior** [comment]
@StefanPlizga
as far as I know, reading from the slave should work and the INFO command should succeed without causing problems.
Maybe the slave log may give some hints on the issue, do you still have the log?

**Comment by StefanPlizga** [comment]
@enricogior 
Yes, I still have the logs, however I can't find this error message in them and I don't recall exactly at what time this issue occured, so I can't narrow down in the logs for the moment. I'll keep an eye and post back when the issue occurs again (and I am sure it will because I experienced this many times) with log file extract.

**Comment by enricogior** [comment]
@StefanPlizga 
sounds good.

**Comment by StefanPlizga** [comment]
Hi @enricogior 
FYI, the exact message I get from redis-cli is "Error: The disk is in use or locked by another process."
I found out why I get this: maxclient number has been reached.

**Comment by enricogior** [comment]
@StefanPlizga 
ok, we will try to fix all those misleading error messages.
Thank you.

**Comment by StefanPlizga** [comment]
Hi @enricogior 
FYI, it looks like reaching the maximum number of client has a negative impact on the stability of Redis: since I increased the maxclient number to a higher one, the Redis instance did not crash at all. This was the only change done since then.
Trace in the last comment from https://github.com/MSOpenTech/redis/issues/310 may be related to this.
